starknet_py/tests/e2e/proxy/proxy_test.py

```diff
@@ -0,0 +1,173 @@
+from typing import Optional
+
+import pytest
+from starkware.starknet.public.abi import get_storage_var_address
+
+from starknet_py.contract import Contract
+from starknet_py.net.client import Client
+from starknet_py.net.client_errors import ContractNotFoundError
+from starknet_py.net.models import Address
+from starknet_py.proxy.contract_abi_resolver import ProxyResolutionError
+from starknet_py.proxy.proxy_check import ProxyCheck
+
+
+async def map_works_properly(map_contract: Contract, key: int, val: int) -> bool:
+    """Put (key, val) into map_contract's storage and check if value under the key is val"""
+    await (
+        await map_contract.functions["put"].invoke(key, val, max_fee=int(1e16))
+    ).wait_for_acceptance()
+    (result,) = await map_contract.functions["get"].call(key=key)
+    return result == val
+
+
+@pytest.mark.asyncio
+async def test_contract_from_address_no_proxy(
+    gateway_account_client, map_compiled_contract
+):
+    deployment_result = await Contract.deploy(
+        compiled_contract=map_compiled_contract,
+        client=gateway_account_client,
+    )
+    await deployment_result.wait_for_acceptance()
+
+    contract = await Contract.from_address(
+        address=deployment_result.deployed_contract.address,
+        client=gateway_account_client,
+    )
+
+    assert contract.functions.keys() == {"put", "get"}
+    assert contract.address == deployment_result.deployed_contract.address
+    assert await map_works_properly(map_contract=contract, key=69, val=13)
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "deploy_proxy_to_contract",
+    [
+        ("oz_proxy_compiled.json", "map_compiled.json"),
+        ("argent_proxy_compiled.json", "map_compiled.json"),
+    ],
+    indirect=True,
+)
```

Comment:
Can't we use a parametrized fixture instead of indirect? It's more confusing this way.